src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -46,108 +54,155 @@
+              masterList[i][0] = currUser.getCreationTime().toString();
+              masterList[i][1] = "u";
+              masterList[i][2] = currUser.getName();
+              masterList[i][3] = "";
+              masterList[i][4] = "";
+          }
 
-      // empty?
-      if(conversations == null || conversations.isEmpty()){
-      %>
-        <p>No New Conversations.</p>
-      <%
-      }
-      else{ // not empty? make a list
-      %>
-        <ul class="mdl-list">
-      <%
-        while(itrC.hasPrevious()){
-          Conversation currConvo = itrC.previous();
-      %>
-        <li>
-          <strong> <%= currConvo.getCreationTime() %>: </strong> 
-          <%= (allUsers.getUser(currConvo.getOwnerId())).getName() %> created a new conversation: 
-          <a href="/chat/<%= currConvo.getTitle() %>"> <%= currConvo.getTitle() %></a>
-        </li> 
-      <%
-        }
-      %>
-        </ul>
-      <%
-      }
-      %>
+          // add convos
+          for (int i = 0; i < conversations.size(); i++) {
 
-      <!-- NEW MESSAGES -->
-      <h3>New Messages</h3>
-     
+              Conversation currConvo = conversations.get(i);
+              int buff = users.size();
 
-    </div>
+              masterList[i+buff][0] = currConvo.getCreationTime().toString();
+              masterList[i+buff][1] = "c";
+              masterList[i+buff][2] = (userStore.getUser(currConvo.getOwnerId())).getName();
+              masterList[i+buff][3] = currConvo.getTitle();
+              masterList[i+buff][4] = "";
+          }
+
+          // // add messages
+          for (int i = 0; i < messages.size(); i++) {
+
+              Message currMessage = messages.get(i);
+              int buff = conversations.size() + users.size();
+              
+              masterList[i+buff][0] = currMessage.getCreationTime().toString();
+              masterList[i+buff][1] = "m";
+              masterList[i+buff][2] = (userStore.getUser(currMessage.getAuthorId())).getName();
+              masterList[i+buff][3] = (convoStore.getConvoWithID(currMessage.getConversationId())).getTitle();
+              masterList[i+buff][4] = currMessage.getContent();
+          }
+
+
+
+          // SORT MASTER LIST BASED ON TIMESTAMP
+          Arrays.sort(masterList, new java.util.Comparator<String[]>() {
+               public int compare(String[] a, String[] b) {
+                   
+                  Instant ins1 = Instant.parse(a[0]);
+                  Date date1 = Date.from(ins1);
 
+                  Instant ins2 = Instant.parse(b[0]);
+                  Date date2 = Date.from(ins2);
 
+                   if (date2.before(date1))
+                      return -1;
+                   else
+                     return 1;
+               }
+           }
+        );
+
+      %>
+
+
+    <div style="background-color:Silver">
+
+      <% 
+          // DATE FORMAT
+          SimpleDateFormat sdf = new SimpleDateFormat("EEE MMM dd HH:mm:ss z yyyy");
```

Comment:
Same here. Move this into the servlet and just pass the final processed data to this template.